tests/Model/IndexInfoTest.php

```diff
@@ -139,4 +139,24 @@ public function testOffsetUnsetCannotBeCalled()
         $this->expectExceptionMessage('MongoDB\Model\IndexInfo is immutable');
         unset($info['v']);
     }
+
+    public function testGetters()
+    {
+        $info = new IndexInfo([
+            'v' => 1,
+            'key' => ['x' => 1],
+            'name' => 'x_1',
+            'ns' => 'foo.bar',
+        ]);
+
+        $this->assertNull($info->get2dSphereIndexVersion());
+        $this->assertNull($info->getBucketSize());
+        $this->assertEquals(IndexInfo::DEFAULT_LANGUAGE, $info->getDefaultLanguage());
+        $this->assertEquals(IndexInfo::DEFAULT_LANGUAGE_OVERRIDE, $info->getLanguageOverride());
+        $this->assertNull($info->getPartialFilterExpression());
+        $this->assertNull($info->getStorageEngineName());
+        $this->assertNull($info->getTextIndexVersion());
+        $this->assertEquals(IndexInfo::DEFAULT_WEIGHT, $info->getWeights());
+        $this->assertFalse($info->isBackground());
+    }
```

Comment:
I think it's good that you test for default values in case elements in the IndexInfo struct are missing, but I also think the tests should cover the cases where the elements are present. In other comments I've shown a few possible cases (for simple ``text``, and ``2dsphere`` indexes), and I think it would be good to add tests cases for these as well.